### ISSUE TITLE
Larger jpg is not rendered

### ISSUE THREAD

**Issue by mb720** [eXeC64/imv#217]
Hi!

I'm having trouble rendering images that are a bit larger. [This one](https://mega.nz/#!L4AUGIia!XUhH7sDY-O8UxuaUyrIJK8Onnk2VMCgvBVdI-qfAjCs) for example is about 28 MiB compressed. When opening it with imv 4.0.1, this is what I see (the teal on the sides is my background color for imv):

![imv screenshot](https://i.imgur.com/u5KbK79.png)

Whereas the image renders fine in GIMP:

![GIMP screenshot](https://i.imgur.com/WeOBkmW.png)

I'm on Wayland (Sway 1.2).

Please let me know if I can provide any more information.

**Comment by rien333** [comment]
Could be related to this: https://github.com/eXeC64/imv/issues/197.

That issue was resolved by increasing the maximum allowed size for pngs. Maybe something similar can be done for jpgs? If you want and there is indeed such a setting, you could try recompiling imv with tweaked values in the jpg backend.

**Comment by mb720** [comment]
Thanks for the hint! I'm not sure it's related to that since imv *tries* to open huge JPGs like [this one](https://en.wikipedia.org/wiki/File:%D0%90%D0%BB%D0%B5%D0%BA%D1%81%D0%B0%D0%BD%D0%B4%D1%80_%D0%90%D0%BD%D0%B4%D1%80%D0%B5%D0%B5%D0%B2%D0%B8%D1%87_%D0%98%D0%B2%D0%B0%D0%BD%D0%BE%D0%B2_-_%D0%AF%D0%B2%D0%BB%D0%B5%D0%BD%D0%B8%D0%B5_%D0%A5%D1%80%D0%B8%D1%81%D1%82%D0%B0_%D0%BD%D0%B0%D1%80%D0%BE%D0%B4%D1%83_(%D0%AF%D0%B2%D0%BB%D0%B5%D0%BD%D0%B8%D0%B5_%D0%9C%D0%B5%D1%81%D1%81%D0%B8%D0%B8)_-_Google_Art_Project.jpg) but it gets killed by Arch Linux since it uses too much RAM.

**Comment by rien333** [comment]
> Thanks for the hint! I'm not sure it's related to that since imv _tries_ to open huge JPGs

Just to be clear: I'm pretty sure that https://github.com/eXeC64/imv/issues/197 says that imv will _try_ to open large pngs, but that they're not shown (so the problem sounds similar to yours).

**Comment by rien333** [comment]
Oh FIY: the jpg from your original post works fine for me (takes about 2s to load though). Our systems might differ a bit, however. I'm running X11, and one of us might be using the `libjpeg-turbo` backend, and the other freeimage backend (I have both installed, and I have no idea which imv uses when both are present)

**Comment by mb720** [comment]
That's interesting, thanks for trying out the image.

`imv -h` tells me that I'm running the FreeImage backend:

    imv v4.0.1
    See manual for usage information.
    This version of imv has been compiled with the following backends:

    Name: libRSVG
    Description: SVG library developed by GNOME
    Website: https://wiki.gnome.org/Projects/LibRsvg
    License: Lesser GNU Public License

    Name: FreeImage
    Description: Open source image library supporting a large number of formats
    Website: http://freeimage.sourceforge.net/
    License: FreeImage Public License v1.0

    imv's full source code is published under the terms of the MIT
    license, and can be found at https://github.com/eXeC64/imv

    imv uses the inih library to parse ini files.
    See https://github.com/benhoyt/inih for details.
    inih is used under the New (3-clause) BSD license.

Is there a way (command line switch) to make imv use a different backend, in order to narrow down the issue, or is it a compile time option?

**Comment by rien333** [comment]
nvm, I'm also running the freeimage backend, so the backend is not the problem  (didn't know `-h` showed you backend info). Nevertheless, if you're desperate, the libjpg turbo backend might worth a try. Think it is compile time only.

**Comment by mb720** [comment]
I'd like to get at the root of this bug. Is there a way I can open an image with libjpg turbo directly to see whether the issue occurs there too?

**Comment by mb720** [comment]
[This image](https://i.imgur.com/9TlyCF0.png) with only 3.82 MiB doesn't render as well (all white in imv).

`file` prints this about the image:

    deps.png: PNG image data, 10528 x 1499, 8-bit/color RGBA, non-interlaced

This tells me that file size on disk doesn't have to be necessarily large.

**Comment by ericonr** [comment]
@mb720 is that image generated by `graphviz`? I'm having issues with a PNG generated by it as well, `imv` just shows a white screen. My browser can open the image, so I don't think it's malformed.

**Comment by mb720** [comment]
@ericonr: Yes, that image was created with graphviz, via [`stack dot`](https://docs.haskellstack.org/en/stable/dependency_visualization/). It shows dependencies in a Haskell project.

**Comment by ericonr** [comment]
I can open the same image with `deepin-image-viewer`, which also seems to use `FreeImage` as the backend. So it's likely an `imv` issue. I tried building it with support for larger mallocs, but didn't see any difference.

**Comment by rien333** [comment]
Do note that I still can't reproduce. The most recent image  you guys posted opens very quickly, and looks okay.

**Comment by mb720** [comment]
@rien333: Do you use X or Wayland?
The issue might have something to do with my using imv via Sway.

**Comment by ericonr** [comment]
@rien333 I hope a [Google Drive link](https://drive.google.com/file/d/1uxNLzf_zeHpC-6Uj7KgmHNl5ZPGJhmbG/view?usp=sharing) is fine. This is the image I am having issues with.

**Comment by rien333** [comment]
I'm using X. Maybe that's something to do with it.


> @rien333 I hope a Google Drive link is fine. This is the image I am having issues with.

As I said, in earlier comments, I can download the image just fine (in all its `10528 x 1499` glory). I still can't reproduce the _problem_ you guys are having, however.

**Comment by mb720** [comment]
If you want to try out Sway for reproducing, I'd suggest doing the following from a virtual console:

  * `pacman -Sy sway` (or the equivalent of your package manager)
  * `sway`
  * <kbd>Windows key</kbd>+<kbd>Enter</kbd> to open a terminal emulator
  * `imv problematic.png`
  * <kbd>Windows key</kbd>+<kbd>e</kbd> to exit Sway

**Comment by ericonr** [comment]
You need to install `alacritty` to use the terminal from the default configuration, unless you're using Debian (I believe they patch it to use xterm instead).

**Comment by mb720** [comment]
You're right, thanks. I just had a look at `/etc/sway/config` on my Arch Linux machine:

    set $term alacritty

**Comment by ericonr** [comment]
Even unsetting `WAYLAND_DISPLAY` and running `imv-x11`, I still get this error.

**Comment by CameronNemo** [comment]
Also seeing a white screen with various semi large JPEG files. Using sway and freeimage backend.

![FHD0077](https://user-images.githubusercontent.com/3212452/96357305-09858a80-10af-11eb-8b2a-52ac5b547e8e.JPG)
